packages/vite-plugin-svelte/src/utils/options.ts

```diff
@@ -101,7 +99,7 @@ export function resolveOptions(
 	viteConfig: ResolvedConfig
 ): ResolvedOptions {
 	const defaultOptions: Partial<Options> = {
-		hot: viteConfig.isProduction ? false : { injectCss: preResolveOptions.emitCss },
+		hot: viteConfig.isProduction ? false : { injectCss: !preResolveOptions.emitCss },
```

Comment:
My bad. This should have `!` based on the previous code. Got lost after the refactor. Otherwise the `enforceOptionsForHmr()` function would make a warning on startup everytime.

Reply:
Removed this change, and moved to #238